### PR TITLE
chore(developer): use kmc messages for kmc-model-info 🗜

### LINKED ISSUE #9305: chore(developer): Add compiler messages for model_info compiler 🗜

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #9305.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc/src/kmlmi.ts [hunk 1]
```diff
@@ -74,16 +74,14 @@ let modelInfoOptions: ModelInfoOptions = {
...
-} catch(e) {
-  console.error(e);
+const data = writeMergedModelMetadataFile(
+  inputFilename,
+  callbacks,
+  modelInfoOptions);
+if(!data) {
   process.exit(SysExits.EX_DATAERR);
 }
+fs.writeFileSync(outputFilename, data);
```
Comment (ermshiperete) [inline developer/src/kmc/src/kmlmi.ts:84]: So we don't need to handle exceptions we get from `fs.writeFileSync`?

File: developer/src/kmc/src/commands/buildClasses/BuildModelInfo.ts [hunk 2]
```diff
@@ -35,27 +36,27 @@ export class BuildModelInfo extends BuildActivity {
 
     const metadata = project.files.find(file => file.getFileType() == KeymanFileTypes.Source.ModelInfo);
     if(!metadata) {
-      // TODO error
-      throw new Error('Missing .model_info file');
+      callbacks.reportMessage(InfrastructureMessages.Error_FileTypeNotFound({ext: KeymanFileTypes.Source.ModelInfo}));
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply developer/src/kmc/src/commands/buildClasses/BuildModelInfo.ts:39]: ah, I see it logs the file extension.

[the remark this answers is not in the document]
Comment (mcdurdin) [reply developer/src/kmc/src/commands/buildClasses/BuildModelInfo.ts:39]: Also, these have different sources -- one is the .model-info compiler, the other is kmc itself, hence InfrastructureMessages, looking for a .model_info file in the project.

File: developer/src/kmc/src/kmlmi.ts (hunk 1 above)
Comment (mcdurdin) [reply developer/src/kmc/src/kmlmi.ts:84]: kmlmi is deprecated so only doing minimal work on it. Replacement, kmc, has much deeper error management.

**Review by ermshiperete — APPROVED** [review]

**Pull request merged**